### PR TITLE
feat/seo-json-ld: enhance articles SEO with Json-LD data

### PR DESCRIPTION
**Pull request by tx0c**
this is so far adding Json-LD to articles page (`/@userName/...-mediaHash`) only

### CODE REVIEW

**Comment by tx0c** [comment]
at local dev it generates this for http://localhost:3000/@935593258/文革实验-为了入戏-这群演员回到了疯狂的1969-zdpuArRLyD1g7voKCJo85beSDZ2JyM9REqKYQo2EwxB7wqQdd

```js
<script type="application/ld+json">
{"@context":"https://schema.org",
 "@type":"Article",
 "headline":"没完没了的批斗，演员真真假假的“表演”，渐渐地大家都入戏了。叶京顺水推舟，将事情加码，从北京派出调查员，将江思远的错误被上升为“江思远事件”。批斗的语言越来越狠，加入批斗的人越来越多，他们将制作好的高帽子挂在江思远头上，并带领大家高喊“打倒江思远”。◇纪录片《入戏》剧照一群入戏很深的演员采访撰文：潘俊文&nbsp;...",
 "image":"https://assets.matters.news/embed/5c10bd27-0c0b-46d3-bf90-3f201cb50778/1552550668-1-副本-jpg",
 "datePublished":"2019-03-14T08:22:51.580Z",
 "dateModified":null,
 "author":{"@type":"Person","name":"潘俊文","url":"https://matters.news/@935593258"}}
</script>
```

in case of articles with no images, do we want to provide a default one?   Google's Guideline also support multiple images if the article has multiple, and Images should be at least 696 pixels wide.

**Comment by tx0c** [comment]
updated:

```json
<script type="application/ld+json">
{"@context":"https://schema.org",
 "@type":"Article",
 "headline":"没完没了的批斗，演员真真假假的“表演”，渐渐地大家都入戏了。叶京顺水推舟，将事情加码，从北京派出调查员，将江思远的错误被上升为“江思远事件”。批斗的语言越来越狠，加入批斗的人越来越多，他们将制作好的高帽子挂在江思远头上，并带领大家高喊“打倒江思远”。◇纪录片《入戏》剧照一群入戏很深的演员采访撰文：潘俊文&nbsp;...",
 "keywords":["电影","中国纪录片","中国独立电影"],
 "datePublished":"2019-03-14T08:22:51.580Z",
 "image":"https://assets.matters.news/embed/5c10bd27-0c0b-46d3-bf90-3f201cb50778/1552550668-1-副本-jpg",
 "author":{"@type":"Person","name":"潘俊文","url":"https://web-develop.matters.news/@935593258"}
}</script>
```

**Review by guoliu — APPROVED** [review]
LGTM 🌲

File: src/views/ArticleDetail/index.tsx [hunk 1]
```diff
@@ -331,9 +334,25 @@ const ArticleDetail = () => {
...
+        jsonLdData={{
+          '@context': 'https://schema.org',
+          '@type': 'Article',
+          headline: summary,
+          /* wordCount: ... */
+          keywords,
+          datePublished: formatISO(Date.parse(article.createdAt)) /* "2015-02-05T08:00:00+08:00", */,
+          dateModified:
+            article.revisedAt ? formatISO(Date.parse(article.revisedAt)) : undefined /* "2015-02-05T09:20:00+08:00", */,
+          image: article.cover || undefined,
```
Comment (tx0c) [inline src/views/ArticleDetail/index.tsx:349]: it's weird that mentioned `Images should be at least 696 pixels wide. Additional image guidelines: Every page must contain at least one image`
https://developers.google.com/search/docs/advanced/structured-data/article#article-types

and it supports an array of images, from our `assets` it looks like a good mapping, but why got an empty `[]` assets from many pictures article
image

Comment (robertu7) [reply src/views/ArticleDetail/index.tsx:349]: Only [the author can access](https://github.com/thematters/matters-server/blob/develop/src/queries/article/assets.ts#L10-L13) the `Article.assets`, it's for [article edit mode ](https://github.com/thematters/matters-web/blob/01636a603f9964f7246f2370e68f96fda737b7b7/src/views/ArticleDetail/EditMode/gql.ts#L19-L21) currently.

If we do need to add the support for JSON-LD, we should revise or add a new field.

**Pull request merged**